src/GridFS/StreamWrapper.php

```diff
@@ -70,6 +76,18 @@ public function getFile(): object
         return $this->stream->getFile();
     }
 
+    /** @return false|array */
+    public function url_stat(string $path, int $flags)
+    {
+        try {
+            $this->stream_open($path, 'r', 0, $openedPath);
+        } catch (FileNotFoundException $e) {
```

Comment:
Noted that this relies on `Bucket::resolveStreamContext()` throwing the exception.

Would `url_stat()` only be used for the basic file access with a registered alias? I'm curious what would happen if this was given an internal file path. Presumably `stream_open()` would fail because a context was not provided and the internal path string cannot be used to obtain a context.

It may be worth making a note of this here.

Reply:
:point_up:

Reply:
Yes, it's called only with `file_exists`, `is_file` and such functions.